lighthouse-core/gather/computed/page-dependency-graph.js

```diff
@@ -163,7 +172,7 @@ class PageDependencyGraphArtifact extends ComputedArtifact {
       let minCandidate = null;
       let minDistance = Infinity;
       // Find the closest request that finished before this CPU task started
-      candidates.forEach(candidate => {
```

Comment:
does this really change things type interference wise? or just drive-by perf/style?

Reply:
yeah, tsc bug (or limitation). `minCadidate` lost its `null | NetworkNode` type after the `forEach()`, and with the if eliminating falsey values it ended up thinking it was `cpuNode.addDependency(never);`